### DISCUSSION TITLE
I am new to this

### DISCUSSION THREAD

**Discussion by faustozamparelli** [joshmedeski/sesh#73] [Q&A]
Hi josh I stumbled across your channel about a month ago when transitioning to nvim and setting up my terminal environment, I want to start off by saying that i appreciate your work very much and i love your content and way of explaining. 
I am fairly new to this tho and I am having some problems setting up this tool.
I have both zoxide running on a fish shell, fzf and tmux installed. I don't really understand the difference between sesh l -t and sesh l -z and for me at least when i do like sesh connect it does not switch to a random directory on my system because it tells me 
❯ sesh connect down
2024/02/22 22:42:43 unable to connect to "down": failed to add "/Users/username/.config/tmux/down" to zoxide: exit status 1

for example if am trying to switch to downloads + i would like to have tmux already opened on my home folder with a session when opening the terminal and then move form there with sesh if possible, before using sesh i had this in my config.fish

 # Open tmux at login
    if not set -q TMUX
        exec tmux -f ~/.config/tmux/tmux.conf new-session -A -s fish
    end

 and it was working really well (basically it created a session if tmux was not running or attached to the one opened if it was) and i was only using windows binded to control+t, then once in the window i was using zoxide to jump where i wanted to jump. 

can someone in the community please help me understand how to set it up. Thanks a lot in advance

**Comment by joshmedeski** [comment]
Hi @faustozamparelli, thanks for the kind words, happy to help!

If you want to ensure sesh connects to exactly what you want, I recommend your command to something like this

```
sesh connect ~/Downloads
```

If you use a partial match (ex: `down`) it will attempt to match a zoxide result, but if you're more specific it should perform better.

What terminal emulator are you using? Perhaps there's a cleaner way to do this with the terminal emulator's confiuration.

**Comment by faustozamparelli** [comment]
I am using Iterm2 with tmux. I thought that sesh was working with zoxide to use its switching capabilities to change dir. That's why i though sesh ch down would work

**Comment by faustozamparelli** [comment]
and i don't understand why after creating a sesh session, if inside it i do tmux kill-server and then do sesh l it still shows it.

**Comment by faustozamparelli** [comment]
maybe i'll wait for a tutorial from you to make it work, thanks a lot for the quick reply tho. Keep going!

**Comment by faustozamparelli** [comment]
i do not have go up and running on my machine, so dumb. I am going to install it and report back. That might be why i am encountering issues

**Comment by joshmedeski** [comment]
Hey @faustozamparelli I reread your issue and maybe this can help.

- `sesh list -t` will list all your tmux sessions (if any exist)
- `sesh list -z` will list all your zoxide results (if any exist)
- In order to add entries to your zoxide database, you have to setup zoxide on your shell and begin to cd into directories. To confirm you are getting zoxide results you can run `zoxide query -l` to list all zoxide results
- I recommend you create a script like `sesh connect (sesh list | fzf)` to select a session, this is how I set up my workflow

My next YouTube video is going to be all about sesh and I'm continuing to update the project's README to help newcomers better understand how to use it.

Thanks again for the helpful feedback, I hope it starts working for you!
